app/services/katello/registration_manager.rb

```diff
@@ -223,9 +223,15 @@ def finalize_registration(host)
         host.subscription_facet.update_from_consumer_attributes(host.subscription_facet.candlepin_consumer.
             consumer_attributes.except(:guestIds, :facts))
         host.subscription_facet.save!
-        host.subscription_facet.update_subscription_status
-        host.content_facet.update_errata_status
-        host.refresh_global_status!
+
+        # we update the status only if build mode is not on.
+        # If it is on we refresh the statuses automatically elsewhere
+        # when build mode is changed from true to false.
+        unless host.build?
```

Comment:
Ooops, this was misguided. We actually never seem to hit this block because we go into build mode *after* subman-register is called :(

the script has:

```
subscription-manager register
register_katello_host
```

Sorry :(

Reply:
Actually, I guess it would work in the non global-reg workflow of setting host to build mode -> restarting to begin the build -> registration happens (skipping status calls) -> build completes and triggers your new method

Reply:
updated :)